### ISSUE TITLE
Proposal: migrate properties to top level field in `configure-director` (breaking change)

### ISSUE THREAD

**Issue by jtarchie** [pivotal-cf/om#310]
# Problem

We were going to add a new property to `configure-director` for the `dns-configuration`. It would've been a top level key for the configuration file.

Example:

```yaml
dns-configuration: {}
iaas-configuration: {}
resource-configuration: {}
# .... some more ....
```

The top level keys of `iaas_configuration`, `director_configuration`, `dns_configuration`, `security_configuration`, and `syslog_configuration` all map to a [single API endpoint](http://docs.pivotal.io/pivotalcf/2-2/opsman-api/#updating-director-and-iaas-properties-experimental) for director properties. But `om` currently flattens out those properties and mixes them in with other configurations that only map to a single endpoint -- ie `resource-configuration`.

This is not future proofed. When a new key is added to the director properties, we've got to extend `om` to support it.

# Proposed Change

We'd like to change `configure-director` config file to have `properties-configuration`. This value would be the payload passed along to director properties. It would remove the need to combine several top level keys into one payload.

The previous above example configuration would transform to:

```yaml
properties-configuration:
  dns_configuration: {}
  iaas_configuration: {}
  director_configuration: {}
resource-configuration: {}
# .... some more ....
```

# Possible migration path

We could pretty easily ship a command that moves the keys in an existing configuration. It would get uglified, but a lot of these are machine-dumped anyway.

**Comment by jtarchie** [comment]
These change have been made and pass CI.

When you encounter this with an older version of the config file, a _helpful error message_ will appear to let you know the changes you need to make.

```bash
$ om --env env/env.yml configure-director --config config/director.yml
could not execute "configure-director": The following keys have recently been removed from the top level configuration: director-configuration, iaas-configuration, security-configuration, syslog-configuration
To fix this error, move the above keys under 'properties-configuration' and change their dashes to underscores.

The old configuration file would contain the keys at the top level.

director-configuration: {}
iaas-configuration: {}
network-assignment: {}
networks-configuration: {}
resource-configuration: {}
security-configuration: {}
syslog-configuration: {}
vm-extensions-configuration: {}

They'll need to be moved to the new 'properties-configuration', with their dashes turn to underscore.
For example, 'director-configuration' becomes 'director_configuration'.

The new configration file will look like.

az-configuration: {}
network-assignment: {}
networks-configuration: {}
properties-configuration:
  director_configuration: {}
  security_configuration: {}
  syslog_configuration: {}
  iaas_configuration: {}
resource-configuration: {}
vm-extensions-configuration: {}
```

When this change goes out in the next release (v0.50.0 at the moment), we'll document the same messaging in the release notes of `om` and `p-automator`.

**Comment by jtarchie** [comment]
At the moment, we've opted-out of automatically doing the migration for you in the file as it would require an additional `output` file to be specified and then be persisted. We don't want to do anything destructive at the moment.

**Comment by jtarchie** [comment]
This has been released in v0.50.0

**Comment by xyloman** [comment]
@jtarchie did this change also update the command `staged-director-config` to return in the new format?
